spki/src/traits.rs

```diff
@@ -123,11 +123,11 @@ where
     T: AssociatedAlgorithmIdentifier,
     T::Params: Into<Any>,
 {
-    fn algorithm_identifier(&self) -> AlgorithmIdentifierOwned {
-        AlgorithmIdentifierOwned {
+    fn algorithm_identifier(&self) -> Result<AlgorithmIdentifierOwned> {
+        Ok(AlgorithmIdentifierOwned {
             oid: T::ALGORITHM_IDENTIFIER.oid,
             parameters: T::ALGORITHM_IDENTIFIER.parameters.map(Into::into),
```

Comment:
With #976 landed you can get rid of the `T::Params: Into<Any>` bound and instead do:

```suggestion
            parameters: T::ALGORITHM_IDENTIFIER.parameters.as_ref().map(Any::encode_from).transpose()?,
```

Reply:
Implemented in a slightly different way.

Reply:
That seems way more verbose, but I guess it gets the job done

Reply:
Maybe it's a matter of preference of simple `if let` compared to long chains of function calls.

Reply:
I imagine clippy will complain about it at some point in the future: https://github.com/rust-lang/rust-clippy/issues/4171

I'd consider the functional approach a bit more idiomatic.

Reply:
that lint talks about iterables. If you prefer, I can use your form, but IMHO it is less clear.

Reply:
If nothing else it's more consistent with the other uses of `transpose()` in this repo

Reply:
done